pkg/router/routeapihelpers/validation.go

```diff
@@ -2,17 +2,32 @@ package routeapihelpers
 
 import (
 	"bytes"
+	"context"
 	"crypto/ecdsa"
 	"crypto/rsa"
 	"crypto/tls"
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
 
-	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/authentication/user"
 	"k8s.io/client-go/util/cert"
 
 	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/library-go/pkg/authorization/authorizationutil"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	kapi "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	authorizationclient "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	// routerServiceAccount is used to validate RBAC permissions for externalCertificate
+	routerServiceAccount = "system:serviceaccount:openshift-ingress:router"
```

Comment:
Can you avoid hard-coding the serviceaccount name?  Maybe using an environment variable as described here: <https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables>.  That might require a change in cluster-ingress-operator, though, so feel free to leave this as a followup and just add a "to-do" comment.

Reply:
Sure, we can use an environment variable to avoid hard coding. I've added a TODO as a follow-up.